### PR TITLE
Added getReserveAddressById

### LINKED ISSUE #589: Add `getReserveAddressById` accessor
*Identified by:*
Aave team

*Scope:*
Add an accessor to fetche the reserve address by id from the Pool contract

### PR DESCRIPTION
**Pull request by The-3D**
#closes #589

### CODE REVIEW

**Comment by The-3D** [comment]
closing this as it is not needed anymore, initial implementation being replaced by #605

**Comment by The-3D** [comment]
reopening as directly requested by the DefiSaver team to be able to decode calldata in external functionc alls

**Review by kartojal — APPROVED** [review]

**Review by 0xgensler — APPROVED** [review]

**Review by LHerskind — APPROVED** [review]
LGTM

File: contracts/interfaces/IPool.sol [hunk 1]
```diff
@@ -602,12 +602,19 @@ interface IPool {
...
    * @dev It does not include dropped reserves
-   * @return The addresses of the reserves
+   * @return The addresses of the underlying assets of the initialized reserves
    **/
   function getReservesList() external view returns (address[] memory);
 
+  /**
+   * @notice Returns the address of the underlying asset of a reserve by the reserve id as stored in the DataTypes.ReserveData struct
+   * @param id The id of the reserve as stored in the DataTypes.ReserveData struct
+   * @return The addresses of the reserve associated with id
```
Comment (miguelmtzinf) [inline contracts/interfaces/IPool.sol:614]: address*

**Pull request merged**